app/react/common/stacks/queries/useStackQuery.ts

```diff
@@ -0,0 +1,47 @@
+import { useQuery } from 'react-query';
+
+import axios from '@/portainer/services/axios';
+import { withError } from '@/react-tools/react-query';
+
+import { Stack, StackFile, StackId } from '../types';
+
+import { stacksQueryKeys } from './query-keys';
+import { buildStackUrl } from './buildUrl';
+
+export function useStackQuery(stackId?: StackId) {
+  return useQuery(
+    stacksQueryKeys.stackFile(stackId || 0),
+    () => getStack(stackId),
+    {
+      ...withError('Unable to retrieve stack'),
+      enabled: !!stackId,
+    }
+  );
+}
+
+async function getStack(stackId?: StackId) {
```

Comment:
this makes the function return `undefined`. I think it should take a defined value.

in the hook, use `!` to state that the id is always defined. because we guard it with `enabled`
```ts
export function useStackQuery(stackId: StackId | undefined) { // use `|undefined` to state that you expect a parameter always. so it should be called `useStackQuery(id)` and to prevent `useStackQuery()`
  return useQuery(
    stacksQueryKeys.stackFile(stackId!),
    () => getStack(stackId!),
    {
      ...withError('Unable to retrieve stack'),
      enabled: !!stackId,
    }
  );
}
```

Reply:
call it `useStack`. we only add `Mutation` prefix for mutations

Reply:
Nice, updated